gobblin-api/src/main/java/org/apache/gobblin/converter/Converter.java

```diff
@@ -120,16 +127,29 @@ public State getFinalState() {
   public RecordStreamWithMetadata<DO, SO> processStream(RecordStreamWithMetadata<DI, SI> inputStream,
       WorkUnitState workUnitState) throws SchemaConversionException {
     init(workUnitState);
-    SO outputSchema = convertSchema(inputStream.getSchema(), workUnitState);
+    this.outputGlobalMetadata = GlobalMetadata.<SI, SO>builderWithInput(inputStream.getGlobalMetadata(),
+        Optional.of(convertSchema(inputStream.getGlobalMetadata().getSchema(), workUnitState))).build();
     Flowable<StreamEntity<DO>> outputStream =
         inputStream.getRecordStream()
             .flatMap(in -> {
               if (in instanceof ControlMessage) {
+                ControlMessage out = (ControlMessage) in;
+
                 getMessageHandler().handleMessage((ControlMessage) in);
-                return Flowable.just(((ControlMessage<DO>) in));
+
+                // update the output schema with the new input schema from the MetadataUpdateControlMessage
+                if (in instanceof MetadataUpdateControlMessage) {
+                  this.outputGlobalMetadata = GlobalMetadata.<SI, SO>builderWithInput(inputStream.getGlobalMetadata(),
```

Comment:
Currently there are no other fields, so it either works. The original idea was to start with the inputStream GlobalMetadata and then overlay from `in`. But I think the suggestion of using `in.getGlobalMetadata()` is good since the copy will be handled in `builderWithInput`.